core/types/data_source_test.go

```diff
@@ -3,14 +3,54 @@ package types_test
 import (
 	"errors"
 	"testing"
+	"time"
 
 	"code.vegaprotocol.io/vega/core/types"
-	"code.vegaprotocol.io/vega/protos/vega"
 	vegapb "code.vegaprotocol.io/vega/protos/vega"
 	datapb "code.vegaprotocol.io/vega/protos/vega/data/v1"
 	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"google.golang.org/protobuf/types/known/structpb"
 )
 
+func TestNewDataSourceDefinitionWith(t *testing.T) {
+	t.Run("oracle", func(t *testing.T) {
+		t.Run("empty", func(t *testing.T) {
+			nds := types.NewDataSourceDefinitionWith(nil)
+			assert.NotNil(t, nds)
+			assert.IsType(t, &types.DataSourceDefinition{}, nds)
+			assert.Nil(t, nds.Content())
+		})
+
+		t.Run("non-empty oracle", func(t *testing.T) {
+			// Do we need to test that?
```

Comment:
@AnNiran do you know, looks like there is still some testing to be done here?

Reply:
That would be a test that is implicitly done in the other places of this file, i was wondering if need to add it.
I'll add it and see if it is too repetitive.

Reply:
```suggestion
```